builtin/logical/database/path_config_connection.go

```diff
@@ -0,0 +1,274 @@
+package database
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/fatih/structs"
+	"github.com/hashicorp/vault/builtin/logical/database/dbplugin"
+	"github.com/hashicorp/vault/helper/strutil"
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+)
+
+var (
+	respErrEmptyPluginName = logical.ErrorResponse("empty plugin name")
+	respErrEmptyName       = logical.ErrorResponse("empty name attribute given")
+)
+
+// DatabaseConfig is used by the Factory function to configure a Database
+// object.
+type DatabaseConfig struct {
+	PluginName string `json:"plugin_name" structs:"plugin_name" mapstructure:"plugin_name"`
+	// ConnectionDetails stores the database specific connection settings needed
+	// by each database type.
+	ConnectionDetails map[string]interface{} `json:"connection_details" structs:"connection_details" mapstructure:"connection_details"`
+	AllowedRoles      []string               `json:"allowed_roles" structs:"allowed_roles" mapstructure:"allowed_roles"`
```

Comment:
I was thinking if, on empty, all were allowed it would let the operator decide if they wanted to control this with acls instead. But they could also just set it to "*" and still add ACL rules, so i'm happy with either option.